### DISCUSSION TITLE
Run away x-axis after pause

### DISCUSSION THREAD

**Discussion by ImSadPanda** [mriscoc/Ender3V2S1#758] [Q&A] [answered]
He everyone, 
I'm just getting into more advanced printing but I keep having a weird issue occur. I use UltiMaker Cura for my prints and nearly every time I insert a filament change pause Z raises up but then instead of X&Y homing my X just heads to the right at full speed and crashes into the end and then grinds the motor for a second and then proceeds with doing a normal filament change sequence. When I resume the print it should be no surprise that X has no idea where its at and starts printing off center of the return point. 

System:

- Ender 3 V2 
- sprite extruder pro. 
- Beagle Camera
- Current firmware of mriscoc
- UltiMaker Cura beta 5.3.0

Attempts to fix: 

1. Bypassed Beagle camera and SD card directly into printer. Fail
2. Adjusted X&Y axis in Cura to slightly different pause points (from 0.0 to 1.0). Fail
3. Re-slicing in Cura multipule times. Fail
4. Restarting Cura and setting up print again. Fail
5. Loaded old Cura 5.2.1 . Success! But pressing pause on the printer, during the print, still causes runaway of X. (Maybe I just got luck on this print? Recall that I said "Nearly every print"). Ugh its 2am I'm going to bed.

**Comment by mriscoc** [comment]
Check your parking position.

**Reply by ImSadPanda** [reply]
Yes, I tried several exact parking locations for X- (0.0, 1.0, 2.0) These were ignored and the hotend still crashes into the far end on the right side away from home. This is looking like a firmware issue. If I give a pause command from my Beagle Camera it works fine. Only when I use the Ender control panel pause button does it run away.

**Comment by mriscoc** [comment, marked as answer]
No issues with pauses in the firmware, your problem could be related to your slicer, check if you are using only M25 for pauses or M600 for filament changes, do not allow your slicer to insert other G-code commands.

**Comment by ImSadPanda** [comment]
I see what your saying. The runaway could be best described as unplanned
pauses. Meaning I'm actually just stopping the printer manually, not G-Code
related at all. When I press pause on my Ender 3v2 manually the x runs away
and crashes into the right side of the printer every time. So I guess it
does it both with code and without code. It just feels heavily like a
firmware issue because it doesn't matter what I'm printing.

**Reply by mriscoc** [reply]
Is your printer connected to a host by USB? A host like Octoprint sends the printer some commands when it detects changes in the printer state like a pause.

**Comment by ImSadPanda** [comment]
Not at this time. I have been using a beagle camera but I had the same
results even with it unplugged.

**Comment by ImSadPanda** [comment]
Hey Miguel,
Have not heard back from you for a minute, so I wanted to give you this
video to show the problem. I have also updated the firmware to the latest
release. This is still occurring.
I did a simple cube print with a filament change at layer 2.  You will see
that the print head crashes into the right side and if you listen it spins
out the stepper motor as it is trying to go beyond the limits of the print
area.
I also tried to use the onboard "filament change" option. I waited for the
print to start, then selected this option and had the same result, print
head crashes into the right side.

Please, If you have any ideas or advice I am all ears!

Thank you
Lance K.
 Ender 3v2 pause problems.mp4
<https://drive.google.com/file/d/13UCAFPmneDvkm0mjotsU1tuqWogyWakW/view?usp=drive_web>

**Reply by mriscoc** [reply]
The video is not accessible (maybe shared privately), but I made some filament changes without issues.

**Comment by ImSadPanda** [comment]
Sorry the video did not play (the file was a bit big and forced a Google
drive link, but that never works for me).
Good news is I finally figured out the problem.
I'll try and explain it in the best way I understand (sorry but i'm a noob
with 3d printing, but I love your firmware! I just don't understand 60% of
it and there's not a FAQ or instruction guide for guys like me... so it's A
LOT of guessing and Ummmmm??? work). So after hunting down all Pause
related subfolders I found that in the firmware you have the default park
position set at x=240.  The standard Ender 3v2 is only 230mm.. I did not
know that I needed to adjust this system setting upon set-up soooo that was
the problem! I reset it to x:0.0, Y:190. It seems to be functioning now
(fingers crossed).

Thank you for always responding to my questions.  I have really appreciated
it.

Ok so here is my newest question with the latest update. In the previous
version, when I used to do a filament change it would do a crazy extrude/
then rapid retract (like fast as hell!) I LOVED IT. However now with the
update it's gone the opposite way with granny slow retract and even slower
Purge...
got a second to explain where I got to go in your settings to put it back
to "normal" (wink)?

Thanks
Lance K

**Reply by mriscoc** [reply]
Allways check the Wiki: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides there you can find the recommendation to check the parking pos.
